tests/appsec/test_automated_user_and_session_tracking.py

```diff
@@ -0,0 +1,187 @@
+# Unless explicitly stated otherwise all files in this repository are licensed under the the Apache License Version 2.0.
+# This product includes software developed at Datadog (https://www.datadoghq.com/).
+# Copyright 2024 Datadog, Inc.
+
+from utils import context
+from utils import features
+from utils import interfaces
+from utils import remote_config as rc
+from utils import rfc
+from utils import scenarios
+from utils import weblog
+
+
+def login_data(context):
+    """In Rails the parameters are group by scope. In the case of the test the scope is user.
+   The syntax to group parameters in a POST request is scope[parameter]
+   """
+    username_key = "user[username]" if "rails" in context.weblog_variant else "username"
+    password_key = "user[password]" if "rails" in context.weblog_variant else "password"
+    return {username_key: "test", password_key: "1234"}
+
+
+@rfc("https://docs.google.com/document/d/1RT38U6dTTcB-8muiYV4-aVDCsT_XrliyakjtAPyjUpw")
+@features.user_monitoring
+class Test_Automated_User_Tracking:
+    def setup_user_tracking_auto(self):
+        self.r_login = weblog.post("/login?auth=local", data=login_data(context))
+        self.r_home = weblog.get("/", cookies=self.r_login.cookies,)
+
+    def test_user_tracking_auto(self):
+        assert self.r_login.status_code == 200
+
+        assert self.r_home.status_code == 200
+        for _, trace, span in interfaces.library.get_spans(request=self.r_home):
+            meta = span.get("meta", {})
+            assert meta["usr.id"] == "social-security-id"
+            assert meta["_dd.appsec.usr.id"] == "social-security-id"
+            assert meta["_dd.appsec.user.collection_mode"] == "ident"
```

Comment:
The possible values for this property are: `sdk`,  `identification` and `anonymizatoin`.
`ident` and `anom` are alias that could be used in configuration, but the tracer should use the complete names.

Reply:
Changed, thanks! (not sure where I got it from :sweat:)